src/main/java/io/cdap/plugin/snowflake/source/batch/SnowflakeBatchSourceConfig.java

```diff
@@ -95,5 +129,11 @@ public void validate(FailureCollector collector) {
       collector.addFailure("Maximum Slit Size cannot be a negative number.", null)
         .withConfigProperty(PROPERTY_MAX_SPLIT_SIZE);
     }
+
+    if (Strings.isNullOrEmpty(importQuery) && Strings.isNullOrEmpty(tableName)) {
```

Comment:
also add validation for `importQueryType`

Reply:
Added bf54c0020a6276a88d2cdafee7f88cd93b3b7904